src/WebJobs.Script.WebHost/Helpers/SemaphoreHelper.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Helpers
+{
+    public class SemaphoreHelper
+    {
+        public virtual bool Wait(SemaphoreSlim semaphore, int millisecondsTimeout)
```

Comment:
Can you clarify why this class is needed? Is this purely to avoid registering a semaphore?

Reply:
Its to make unit testing easier.